src/pybind/mgr/devicehealth/module.py

```diff
@@ -105,23 +105,21 @@ def handle_command(self, _, cmd):
         self.log.error("handle_command")
 
         if cmd['prefix'] == 'device query-daemon-health-metrics':
-            who = cmd.get('who', '')
-            if who[0:4] != 'osd.':
-                return -errno.EINVAL, '', 'not a valid <osd.NNN> id'
-            osd_id = who[4:]
+            (daemon_type, daemon_id) = cmd.get('who', '').split('.')
+            if daemon_type not in ('osd', 'mon'):
+                return -errno.EINVAL, '', 'not a valid daemon name'
```

Comment:
Extend this string to advise that only osd and mon are allowed?  I can imagine occasional confused users being confused enough to try use this on an MDS.